### PR TITLE
[language][move] Added typing tests [part 3]

### PR DESCRIPTION
**Pull request by tnowacki**
##  Motivation

- Added tests for typing pass.
- Tried to cover +/- cases for each check. Covers:
  - Binary ops
  - Unary ops
  - Break/continue
  - pack
  - borrow local

## Test Plan

ran em

### CODE REVIEW

File: language/move-lang/tests/move_check/typing/binary_add_invalid.exp [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+   ┌── tests/move_check/typing/binary_add_invalid.move:8:9 ───
+   │
+ 8 │         false + true;
+   │         ^^^^^ Invalid argument to '+'
+   ·
+ 8 │         false + true;
+   │         ----- Found: 'bool'. But expected: 'u64'
+   │
+
```
Comment (tzakian) [inline language/move-lang/tests/move_check/typing/binary_add_invalid.exp]: Why no `Invalid argument` for `true`?

File: language/move-lang/tests/move_check/typing/binary_add_invalid.exp [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+   ┌── tests/move_check/typing/binary_add_invalid.move:9:9 ───
+   │
+ 9 │         1 + false;
+   │             ^^^^^ Invalid argument to '+'
+   ·
+ 9 │         1 + false;
+   │             ----- The type: 'bool'
+   ·
+ 9 │         1 + false;
+   │         - Is is not compatible with: 'u64'
```
Comment (tzakian) [inline language/move-lang/tests/move_check/typing/binary_add_invalid.exp]: double "is"

File: language/move-lang/tests/move_check/typing/binary_add_invalid.exp (hunk 1 above)
Comment (tnowacki) [reply language/move-lang/tests/move_check/typing/binary_add_invalid.exp]: meh

File: language/move-lang/tests/move_check/typing/binary_add_invalid.exp (hunk 2 above)
Comment (tnowacki) [reply language/move-lang/tests/move_check/typing/binary_add_invalid.exp]: Oh no...

File: language/move-lang/src/typing/core.rs [hunk 3]
```diff
@@ -190,33 +190,52 @@ impl Context {
...
+            };
+            let res = tyl
+                .into_iter()
+                .zip(contraints)
+                .filter_map(|(t, constraint_opt)| {
+                    match (infer_kind_base(context, subst, t), constraint_opt) {
+                        (Some(k), _) | (None, Some(k)) => Some(k),
+                        (None, None) => None,
+                    }
+                })
```
[the remark this answers is not in the document]
Comment (tnowacki) [reply language/move-lang/src/typing/core.rs]: I want the location of the inferred type if its there

[the remark this answers is not in the document]
Comment (tnowacki) [reply language/move-lang/src/typing/core.rs]: Also, `and` does not do what I want here. It returns the second option if the first is `Some`

[the remark this answers is not in the document]
Comment (tnowacki) [reply language/move-lang/src/typing/core.rs]: So I think I want `or`

[the remark this answers is not in the document]
Comment (tzakian) [reply language/move-lang/src/typing/core.rs]: Gaddamit...sorry about that

**Pull request closed without merging**